DetectDuplicatesFSharp/Program.fs

```diff
@@ -0,0 +1,81 @@
+﻿open System.IO
+open System.Linq
+open System.Text.RegularExpressions
+
+type FileReader = string -> Result<string seq,exn>
+
+let output (data:seq<string*string>) =
+    data 
+    |> Seq.iter (fun (key,value) -> printfn "Duplicate %s found in %s" key value)
+
+let getPath (args: string []) =
```

Comment:
I would write this as:
```fsharp
let getPath (args: string []) =
        args |> Array.tryHead |> Option.defaultWith Directory.GetCurrentDirectory
```
or pattern match the `Option` returned from `tryHead`. I would avoid `.Any()` and LINQ, as it doesn't feel idiomatic, is almost always a module function to do what you want (for example `Array.isEmpty`)